### PR TITLE
Add Overall tracking of stages and roles of applicants

### PR DESCRIPTION
**Pull request by xinen26**
For easier reference, this is how HRConnect looks like now.
Screenshot 2024-04-01 at 11 08 10 PM

Bug 1 Need to manually refresh the overview page EDIT: fixed
Bug 2 Entering list and filter commands will not unselect the buttons of stages on filter EDIT: fixed

Yet to add test case, will do so as soon as possible.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/EditApplicantCommand.java [hunk 1]
```diff
@@ -112,7 +111,12 @@ public CommandResult execute(Model model) throws CommandException {
         }
 
         model.setPerson(personToEdit, editedApplicant);
-        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+        //model.updateFilteredRoles();
+        model.updateCount("final_stage");
```
Comment (xinen26) [inline src/main/java/seedu/address/logic/commands/EditApplicantCommand.java:115]: The stages are "final_stage", "initial_application", "Technical Assessment" and "Interview". Seems messy so I will go and standardize the stages and update it accordingly throughout

File: src/main/java/seedu/address/logic/commands/FindCommand.java [hunk 2]
```diff
@@ -29,7 +29,7 @@ public FindCommand(NameContainsKeywordsPredicate predicate) {
     @Override
     public CommandResult execute(Model model) {
         requireNonNull(model);
-        model.updateFilteredPersonList(predicate);
+        model.filterPersonList(predicate);
```
Comment (xinen26) [inline src/main/java/seedu/address/logic/commands/FindCommand.java:32]: Basically, now the Find command finds amongst the FilteredPersonList. So if "Interview" button is selected, it only finds amongst those in "Interview". Let me know if the initial way is preferred, ie find amongst all applicants

File: src/main/java/seedu/address/logic/commands/EditApplicantCommand.java [hunk 3]
```diff
@@ -112,7 +111,12 @@ public CommandResult execute(Model model) throws CommandException {
         }
 
         model.setPerson(personToEdit, editedApplicant);
-        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+        //model.updateFilteredRoles();
```
Comment (xinen26) [inline src/main/java/seedu/address/logic/commands/EditApplicantCommand.java:114]: Commented out code and untidy code will get removed soon

**Comment by ys112** [comment]
@lihaoquan You might want to take a look at the comments.

**Comment by lihaoquan** [comment]
image
The UI seems to be the same as the old UI when I checked out this PR

**Review by YuvBindal — COMMENTED** [review]
Are we supposed to run a filter command based on the stages when a person clicks the respective UI for it?

**Review by YuvBindal — COMMENTED** [review]
LGTM

**Comment by xinen26** [comment]
@lihaoquan hello, can i have your addressbook.json to run? thanks!

**Comment by xinen26** [comment]
@YuvBindal hi, filter command will not be run after someone clicks the button but the code behind it is very similar

File: src/main/java/seedu/address/ui/PersonListPanel.java [hunk 4]
```diff
@@ -43,6 +43,7 @@ protected void updateItem(Person person, boolean empty) {
                 setText(null);
                 return;
             }
+            System.out.println("this updateItem is called");
```
Comment (ys112) [inline src/main/java/seedu/address/ui/PersonListPanel.java:46]: Consider removing this or use logger for more verbose display.

**Review by ys112 — APPROVED** [review]
LGTM. Just a nit mentioned in comment below. Seeing how you have added the new boolean array, would the import command get impacted too?

File: src/main/java/seedu/address/ui/PersonListPanel.java (hunk 4 above)
Comment (xinen26) [reply src/main/java/seedu/address/ui/PersonListPanel.java:46]: Good catch! Will remove it later

**Comment by xinen26** [comment]
@ys112 hello yes i think there might be some merge conflicts, will deal with it before pushing again

**Pull request merged**